File: hash_table/0506_relative_ranks/0506_relative_ranks.go
package main

import (
	"fmt"
	"sort"
	"strconv"
)

func findRelativeRanks(nums []int) []string {
	list := make([]int, len(nums))
	copy(list, nums)
	sort.Slice(list, func(i int, j int) bool {
		return list[i] > list[j]
	})
	dict := make(map[int]int)
	for i, v := range list {
		dict[v] = i + 1
	}
	res := make([]string, len(nums))
	for i := 0; i < len(res); i++ {
		if dict[nums[i]] == 1 {
			res[i] = "Gold Medal"
		} else if dict[nums[i]] == 2 {
			res[i] = "Silver Medal"
		} else if dict[nums[i]] == 3 {
			res[i] = "Bronze Medal"
		} else {
			res[i] = strconv.Itoa(dict[nums[i]])
		}
	}
	return res
}

func main() {
	nums := []int{5, 4, 3, 2, 1}
	res := findRelativeRanks(nums)
	fmt.Println(res)
}

File: hash_table/0506_relative_ranks/0506_relative_ranks.py
from typing import List

class Solution:
    def findRelativeRanks(self, nums: List[int]) -> List[str]:
        hashmap = {}
        ans = [0] * len(nums)
        for i in range(len(nums)):
            hashmap[nums[i]] = i
        nums.sort(reverse = True)
        for i in range(len(nums)):
            if i == 0:
                ans[hashmap[nums[i]]] = 'Gold Medal'
            if i == 1:
                ans[hashmap[nums[i]]] = 'Silver Medal'
            if i == 2:
                ans[hashmap[nums[i]]] = 'Bronze Medal'
            if i > 2:
                ans[hashmap[nums[i]]] = str(i + 1)
        return ans



nums = [5,4,3,2,1]
res = Solution().findRelativeRanks(nums)
print(res)